### ISSUE TITLE
cypress/unsafe-to-chain-command shouldn't complain if I'm not using the subject

### ISSUE THREAD

**Issue by crfrolik** [cypress-io/eslint-plugin-cypress#268]
e.g.:

```typescript
cy.get('foo')
  .click()
  .then(() => someFuncThatDoesNotUseFoo());
```

This is perfectly safe, even though the linter will complain about it. Splitting it out like this adds no value:

```typescript
cy.get('foo')
  .click();
cy.then(() => someFuncThatDoesNotUseFoo());
```
